examples/java-matter-controller/java/src/com/matter/controller/commands/common/Argument.java

```diff
@@ -33,59 +33,78 @@ public final class Argument {
   private final long mMax;
   private final Object mValue;
   private final Optional<String> mDesc;
+  private final boolean mOptional;
 
-  public Argument(String name, IPAddress value) {
+  boolean isOptional() {
+    return mOptional;
+  }
+
+  public Argument(String name, IPAddress value, boolean optional) {
     this.mName = name;
     this.mType = ArgumentType.ADDRESS;
     this.mMin = 0;
     this.mMax = 0;
     this.mValue = value;
     this.mDesc = Optional.empty();
+    this.mOptional = optional;
   }
 
-  public Argument(String name, StringBuffer value, @Nullable String desc) {
+  public Argument(String name, StringBuffer value, @Nullable String desc, boolean optional) {
```

Comment:
at some point we should move to a builder pattern - having so many constructors and various arguments starts making the code hard to read.

Reply:
Acked: https://github.com/project-chip/connectedhomeip/issues/23732